dspace-api/src/main/java/org/dspace/app/util/DCInput.java

```diff
@@ -16,9 +18,14 @@
 
 import org.apache.commons.lang3.StringUtils;
 import org.dspace.content.MetadataSchemaEnum;
+import org.dspace.core.Context;
 import org.dspace.core.Utils;
+import org.json.JSONArray;
+import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.util.ObjectUtils;
+import org.xml.sax.SAXException;
 
```

Comment:
CONFIG:
- how does one configure autocomplete for a field
- regexp support seems to be implemented in upstream, is it possible to configure custom error messages where the validation fails?
- can you still add rendering hints? originally `String extraClass` (element `class` in the input-forms.xml)
```
<class>with-datepicker</class>
<class>with-datepicker</class>
<class>no-thumbs</class>
<class>with-datepicker</class>
<class>no-thumbs hide-me</class>
```
- dtto for `class` of a complex type

Reply:
1. how configure autocomplete for a field: https://github.com/dataquest-dev/DSpace/blob/dtq-dev/dspace/config/submission-forms.xml#L88
2. Error messages are defined inside the `<required>ERROR_MESSAGE</req..>`: https://github.com/dataquest-dev/DSpace/blob/dtq-dev/dspace/config/submission-forms.xml#L107
3. Hints are defined inside the `<hint></hing>` tag. I don't know what do you mean with the `rendering hints`. I think it is not implemented.

Reply:
Ad 3. the content of the class element was added into the class attribute of the html element belonging to the field, and during submission, css, js, or even the xslt could do something based on the class. So here https://github.com/ufal/clarin-dspace/blob/8cf3758e9f12c18a2b8a20c13d63757d8ee2d52e/dspace/config/input-forms.xml#L120 the `with-datepicker` class is added. Then in submission, the html input element has these `class="form-control submit-text with-datepicker"` .

I guess I don't need the with-datepicker (a calendar is shown) there's the input type date and the date component.
Class `no-thumbs` is used on `dc.type`; if you are submitting a language resource, you see icons (thumbnails) instead of a dropdown; in other cases, we need the dropdown.
How would you do that now? Two different input-types?

Class `hide-me` is used on dc.type too. In a submission with only one possible dc.type, we just hide it from the user but still get a value.

Reply:
OK I think I understand what do you mean. We are talking about this option? 
![image](https://user-images.githubusercontent.com/90026355/218742579-48904b32-1791-4b44-8a0b-8187b014a628.png)

That `class` tag functionality could be done in the FE e.g., https://github.com/dataquest-dev/dspace-angular/blob/dtq-dev/src/app/shared/form/builder/parsers/onebox-field-parser.ts#L45 
I would check if the metadatavalue of that input type id `dc.type` and add custom `class` there e.g. `grid: {
          host: 'col-sm-8 pl-0 no-thumbs'
        }` 
will generate `class="col-sm-8 pl-0 no-thumbs"`